src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -337,8 +337,12 @@ public static boolean verifyDevice(TenantIdentifierWithStorage tenantIdentifierW
         // verified in the devices table (because it was deleted/renamed). So the user
         // gets a UnknownDevceException.
         // This behaviour is okay so we can ignore it.
-        checkAndStoreCode(tenantIdentifierWithStorage, main, userId, new TOTPDevice[]{matchingDevice},
-                code);
+        try{
+            checkAndStoreCode(tenantIdentifierWithStorage, main, userId, new TOTPDevice[]{matchingDevice}, code);
+        } catch (TotpNotEnabledException e) {
```

Comment:
This can raise `TotpNotEnabledException` only if device was deleted in between. So should we throw `UnknownDeviceError` and just get rid of `TotpNotEnabledException` completely?

Reply:
we should throw UnknownDeviceError and get rid of TotpNotEnabledException entirely.

Reply:
Done